### DISCUSSION TITLE
magit-git-command doesn't honour Tramp path or magit-remote-git-executable

### DISCUSSION THREAD

**Discussion by bjm9991** [magit/magit#5269] [Support]
Output of `M-x magit-version RET`:
```
Magit 20241217.2208 [>= 20241217.2208], Transient 20241218.1310, Forge 20241208.2219, Git 2.39.5, Emacs 29.4, darwin
```

Sometimes it is necessary to use `magit-git-command` to run an arbitrary Git command that is not handled natively by Magit. However, when in a remote Tramp session, this seems to not honour the configured Tramp path or `magit-remote-git-executable`.

Using "standard" Magit commands, seem to work fine. In fact, `magit-version` returns the correctly configured Git version as pointed to by `magit-remote-git-executable`. But, when using `magit-git-command --version`, it returns a much older version which is installed in the default PATH on the remote machine.

**Comment by tarsius** [comment]
`magit-git-command` uses the same low-level function as `magit-shell-command`. 

That low level function, `magit--shell-command`, in turn uses `magit-start-process`. Both of these functions know nothing about `git`, they do not see `git` and replace that with `/path/to/git`.

In other words, it is expected that `magit-remote-git-executable` is ignored.

Now what about "the configured Tramp path". I don't really know because I don't use Tramp. `magit-start-process` ultimately calls `start-file-process` and if that is supposed to pick up "the configured Tramp path", then its beyond the scope of Magit to figure out why it apparently doesn't do that for you in this case.

**Comment by bjm9991** [comment]
Okay, this was probably my bad - I was missing `(with-eval-after-load 'tramp ...` which was causing my `tramp-remote-path` updates to be overwritten.
